src/cogent3/core/new_alignment.py

```diff
@@ -4343,14 +4389,27 @@ def no_degenerates(self, motif_length: int = 1, allow_gap: bool = False):
             )
             raise ValueError(msg) from e
 
-        if allow_gap:
-            chars.extend(self.moltype.gap)
+        array_pos = self.array_positions
+        cutoff = len(chars) + 1 if allow_gap else len(chars)
```

Comment:
add a comment before this definition of cutoff stating that, by design, char alphabets are organised such that the canonical characters always occur first, followed by gap, then ambiguity characters.